MyThreadPool/MyThreadPool/MyThreadPool.cs

```diff
@@ -0,0 +1,87 @@
+// <copyright file="MyThreadPool.cs" company="Gorlov Kirill">
+// Copyright (c) Gorlov Kirill. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the repository root for license information.
+// https://github.com/GolrovKirill/SPbU-Prog3/blob/main/LICENSE
+// </copyright>
+
+namespace MyThreadPool;
+
+/// <summary>
+/// Class that implements thread pool.
+/// </summary>
+public class MyThreadPool
+{
+    private readonly CancellationTokenSource clt = new();
+
+    private readonly Queue<Action> queue = new();
+
+    private readonly Thread[] threads;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MyThreadPool"/> class.
+    /// </summary>
+    /// <param name="countThreads">Number of threads.</param>
+    public MyThreadPool(int countThreads)
+    {
+        ArgumentOutOfRangeException.ThrowIfNegativeOrZero(countThreads);
+
+        threads = new Thread[countThreads];
+        for (var i = 0; i < countThreads; i++)
+        {
+            threads[i] = new Thread(ExecuteTasks);
```

Comment:
```suggestion
            this.threads[i] = new Thread(ExecuteTasks);
```
нужно дописать this везде, где используется поле threads, а то StyleCop ругается